File: test/f90/IsMomentRegistredZC_39.f90
MODULE IsMomentRegistredZC_39Module
    USE Crop2mlModules
    IMPLICIT NONE
CONTAINS
     SUBROUTINE Calculate_IsMomentRegistredZC_39(calendarMoments, isMomentRegistredZC_39)
        !      INPUT VARIABLES:
        CHARACTER(LEN=*),DIMENSION(:), INTENT(IN):: calendarMoments
        !     OUTPUT VARIABLES:
        INTEGER , INTENT(OUT):: isMomentRegistredZC_39

        IF (ANY(calendarMoments=="FlagLeafLiguleJustVisible")) THEN
           isMomentRegistredZC_39 = 1
        ELSE
            isMomentRegistredZC_39 = 0
        ENDIF
    END SUBROUTINE Calculate_IsMomentRegistredZC_39
END MODULE IsMomentRegistredZC_39Module

File: src/cpp/source/IsMomentRegistredZC_39.cpp
#include <iostream>
#include <vector>
#include<string>
#include<algorithm>
#include "IsMomentRegistredZC_39.h"

using namespace std;
IsMomentRegistredZC_39 Calculate_IsMomentRegistredZC_39(vector<string> calendarMoments){

int isMomentRegistredZC_39;
size_t posFlag = find(calendarMoments.begin(), calendarMoments.end(),"FlagLeafLiguleJustVisible" ) - calendarMoments.begin();

if (posFlag<calendarMoments.size())
{
   isMomentRegistredZC_39 = 1 ;
}
else
    isMomentRegistredZC_39 = 0;
return IsMomentRegistredZC_39{isMomentRegistredZC_39};
}

